dev/pl011/uart.c

```diff
@@ -37,24 +37,22 @@
 #define NUM_UART 1
 
 static cbuf_t uart_rx_buf[NUM_UART];
+static uintptr_t uart_base[NUM_UART];
 
 static inline uintptr_t uart_to_ptr(unsigned int n) {
-    switch (n) {
-        default:
-        case 0:
-            return UART_BASE;
-    }
+    return uart_base[n];
 }
 
 static enum handler_return uart_irq(void *arg) {
     bool resched = false;
-    uint port = (uintptr_t)arg;
+    uint port = (uint)arg;
```

Comment:
This is an issue with 64bit targets, since its down casting to a smaller size. Perhaps cast it through uintptr_t first, then down to uint.